src/Symfony/Bridge/Doctrine/Tests/Form/Type/EntityTypeTest.php

```diff
@@ -1516,4 +1516,37 @@ public function testSubmitNullExpandedMultiple()
         $this->assertEquals($collection, $form->getNormData());
         $this->assertSame(array(), $form->getViewData(), 'View data is always an array');
     }
+
+    public function testSetDataEmptyArraySubmitNullMultiple()
+    {
+        $emptyArray = array();
+
+        $form = $this->factory->create(static::TESTED_TYPE, null, array(
+            'em' => 'default',
+            'class' => self::SINGLE_IDENT_CLASS,
+            'multiple' => true,
+        ));
+        $form->setData($emptyArray);
+        $form->submit(null);
+
+        $this->assertInternalType('array', $form->getData());
```

Comment:
The expected data should be `ArrayCollection()` for both cases, see other tests https://github.com/stoccc/symfony/blob/ead368583a9c67c98894bab5502574fd9865a8d4/src/Symfony/Bridge/Doctrine/Tests/Form/Type/EntityTypeTest.php#L377